pkg/controllers/allocation/controller.go

```diff
@@ -66,7 +66,16 @@ func NewController(kubeClient client.Client, coreV1Client corev1.CoreV1Interface
 // Reconcile executes an allocation control loop for the resource
 func (c *Controller) Reconcile(ctx context.Context, object client.Object) (reconcile.Result, error) {
 	provisioner := object.(*v1alpha2.Provisioner)
-	// 1. Filter pods
+
+	// 1. Complement provisioner with (dynamic) default values, which must not
```

Comment:
I might say Hydrate instead of Complement. I'd also be explicit that they are not persisted, e.g. "must not be persisted into ..."